src/serialbox/core/CMakeLists.txt

```diff
@@ -7,42 +7,68 @@
 ##
 ##===------------------------------------------------------------------------------------------===##
 
+include(yodaCreateLibrary)
 cmake_minimum_required(VERSION 3.1)
 
 set(SOURCES 
   FieldMap.cpp
+  FieldMap.h
   FieldMetainfoImpl.cpp
+  FieldMetainfoImpl.h
   FieldID.cpp
+  FieldID.h
   Logging.cpp
+  Logging.h
   MetainfoMapImpl.cpp
+  MetainfoMapImpl.h
   MetainfoValueImpl.cpp
+  MetainfoValueImpl.h
   SavepointImpl.cpp
+  SavepointImpl.h
   SavepointVector.cpp
+  SavepointVector.h
   SerializerImpl.cpp
+  SerializerImpl.h
   StorageView.cpp
+  StorageView.h
   Type.cpp
+  Type.h
   Unreachable.cpp
+  Unreachable.h
   
   hash/HashFactory.cpp
+  hash/HashFactory.h
   hash/SHA256.cpp
+  hash/SHA256.h
   hash/MD5.cpp
+  hash/MD5.h
   
   archive/ArchiveFactory.cpp
+  archive/ArchiveFactory.h
   archive/BinaryArchive.cpp
+  archive/BinaryArchive.h
   archive/NetCDFArchive.cpp
+  archive/NetCDFArchive.h
   archive/MockArchive.cpp
+  archive/MockArchive.h
   
   frontend/stella/MetainfoSet.cpp
+  frontend/stella/MetainfoSet.h
   frontend/stella/Savepoint.cpp
+  frontend/stella/Savepoint.h
   frontend/stella/Serializer.cpp
+  frontend/stella/Serializer.h
   frontend/stella/DataFieldInfo.cpp
+  frontend/stella/DataFieldInfo.h
 )
 
-set(EXTERNAL_LIBRARIES ${Boost_LIBRARIES} ${SERIALBOX_EXTERNAL_LIBRARIES})
 
-serialbox_add_library(NAME SerialboxCore 
-                      TARGET SerialboxCoreLibrary 
-                      BUILD_STATIC
-                      BUILD_SHARED
+set(EXTERNAL_LIBRARIES ${Boost_LIBRARIES} ${STD_FILESYSTEM_LIB} ${CMAKE_THREAD_LIBS_INIT})
+yoda_create_library(TARGET SerialboxCore
                       LIBRARIES ${EXTERNAL_LIBRARIES}
-                      SOURCES ${SOURCES})
+                      SOURCES ${SOURCES}
+                      PUBLIC_BUILD_INCLUDES ${CMAKE_SOURCE_DIR}/src SYSTEM ${Boost_INCLUDE_DIRS}

```

Comment:
This is yet another example of `yoda_create_library`... but do you see the difference? 
A small difference is the use here of the keyword `SYSTEM` which is worth clarifying...
Well, yoda supports here prepending the include paths with the keyword `SYSTEM` that is by cmake to pass to the compiler the header using the `-isystem` flag, which will have the effect of ignoring all the warnings (since we consider we dont care, since Boost is not our library) . 
But the interesting difference here, is that we dont populate the `INSTALL` interface for the the `Boost` include directory.
The reason is that is not recommended by CMake, since once we install it and other projects import ours, they might be using different boost installed on a different path. Therefore system dependencies are recommended to be populated for the `BUILD` interface only. 
Notice also that while building any target that depends on `SerialboxCore` will get the propagated include directories of Boost. But not in the case the project is imported from the install. 

https://cmake.org/cmake/help/v3.3/prop_tgt/INTERFACE_INCLUDE_DIRECTORIES.html